checks/pinned_dependencies.go

```diff
@@ -121,87 +121,123 @@ func maxScore(s1, s2 int) int {
 	return s2
 }
 
-func createReturnValues(r bool, infoMsg string, dl checker.DetailLogger, err error) (int, error) {
+type pinnedResult int
+
+const (
+	pinnedUndefined pinnedResult = iota
+	pinned
+	notPinned
+)
+
+func addPinnedResult(r *pinnedResult, to bool) {
+	// If the result is `notPinned`, we keep it.
+	// In other cases, we always update the result.
+	if *r == pinnedUndefined || *r == pinned {
+		switch to {
+		case true:
+			*r = pinned
+		case false:
+			*r = notPinned
+		}
+	}
+}
+
+func dataAsResultPointer(data FileCbData) *pinnedResult {
+	pdata, ok := data.(*pinnedResult)
+	if !ok {
+		// This never happens.
+		panic("invalid type")
+	}
+	return pdata
+}
+
+func createReturnValues(r pinnedResult, infoMsg string, dl checker.DetailLogger, err error) (int, error) {
 	if err != nil {
 		return checker.InconclusiveResultScore, err
 	}
-	if !r {
+
+	switch r {
+	default:
```

Comment:
Nit: make `default` as the last case in switch statement

Reply:
is this a golang rule? I found that having it first is less error-prone because it prevents forgetting it.

Reply:
I don't think so. I'm just used to seeing `default` at the end because of C++ :) Feel free to leave it as is if you think this improves readability.